tests/drivers/build_all/app.overlay

```diff
@@ -0,0 +1,1035 @@
+				spi-max-frequency = <0>;
+				int-gpios = <&test_gpio 0 0>;
+			};
+
+			mmc-spi-slot@2 {
+				compatible = "zephyr,mmc-spi-slot";
+				label = "MMC-SPI-SLOT";
+				reg = <0x2>;
+				spi-max-frequency = <0>;
+			};
+
+			lmp90100@3 {
+				compatible = "ti,lmp90100";
+				label = "LMP90100";
+				reg = <0x3>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90097@4 {
+				compatible = "ti,lmp90097";
+				label = "LMP90097";
+				reg = <0x4>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90077@5 {
+				compatible = "ti,lmp90077";
+				label = "LMP90077";
+				reg = <0x5>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90078@6 {
+				compatible = "ti,lmp90078";
+				label = "LMP90078";
+				reg = <0x6>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90098@7 {
+				compatible = "ti,lmp90098";
+				label = "LMP90098";
+				reg = <0x7>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90079@8 {
+				compatible = "ti,lmp90079";
+				label = "LMP90079";
+				reg = <0x8>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90080@9 {
+				compatible = "ti,lmp90080";
+				label = "LMP90080";
+				reg = <0x9>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			lmp90099@a {
+				compatible = "ti,lmp90099";
+				label = "LMP90099";
+				reg = <0xa>;
+				spi-max-frequency = <0>;
+				drdyb-gpios = <&test_gpio 0 0>;
+				#io-channel-cells = <2>;
+			};
+
+			bt-hci-spi@b {
+				compatible = "zephyr,bt-hci-spi";
+				label = "BT-HCI-SPI";
+				reg = <0xb>;
+				spi-max-frequency = <0>;
+				irq-gpios = <&test_gpio 0 0>;
+				reset-gpios = <&test_gpio 0 0>;
+			};
+
+			sx1276@c {
+				compatible = "semtech,sx1276";
+				label = "SX1276";
+				reg = <0xc>;
+				spi-max-frequency = <0>;
+				reset-gpios = <&test_gpio 0 0>;
+				dio-gpios = <&test_gpio 0 0>;
+			};
+
+			spi-nor@d {
+				compatible = "jedec,spi-nor";
+				label = "SPI-NOR";
+				reg = <0xd>;
+				spi-max-frequency = <0>;
+				wp-gpios = <&test_gpio 0 0>;
+				hold-gpios = <&test_gpio 0 0>;
+				reset-gpios = <&test_gpio 0 0>;
+				jedec-id = [];
+				/* has-be32k; */
+				/* requires-ulbpr; */
+				/* has-dpd; */
+				size = <0>;
+			};
+
+			w25q16@e {
+				compatible = "winbond,w25q16";
+				label = "W25Q16";
+				reg = <0xe>;
+				spi-max-frequency = <0>;
+			};
+
+			at25@f {
+				compatible = "atmel,at25";
+				label = "AT25";
+				reg = <0xf>;
+				spi-max-frequency = <0>;
+				size = <32768>;
+				pagesize = <64>;
+				address-width = <16>;
+				timeout = <5>;
+				wp-gpios = <&test_gpio 0 0>;
+				/* read-only; */
+			};
+
+			mcr20a@10 {
```

Comment:
Did you say you generated this overlay with a script? There seem to be additional nodes that weren't previously in the test. Can we keep the nodes in alpha order?

Reply:
I had it general all I2C or SPI client devices, thus we may have more than was in dts_fixup.h.  While we can sort it alphabetically, the issue is that the reg value needs to be unique.  So lets say we have:

node_a / reg = 1
node_c / reg = 2

what happens when we add node_b, what reg value would it have?

The dts_fixup.h used the same reg value for everything, but when we build with DTS we'll get a DTS compile warning if we have the same reg value.

Reply:
Would it make sense to put each device node on its own vnd,i2c or vnd,spi bus? Then the device regs would be independent. The addresses of the buses don't matter as long as they're different.

bus-1 { node_a / reg = 1 }
bus-2 { node_c / reg = 1 }

---->

bus-1 { node_a / reg = 1 }
bus-2 { node_b / reg = 1 }
bus-3 { node_c / reg = 1 }

Reply:
@mbolivar-nordic while we can do that, not sure what benefit that creates.  It means whenever someone adds a new device they're going to have to add a new controller which seems worse to me, than just having them add the device at the end of the list.